Realm/RLMArray.mm

```diff
@@ -299,6 +299,25 @@ - (RLMResults *)objectsWhere:(NSString *)predicateFormat args:(va_list)args
     return [self objectsWithPredicate:[NSPredicate predicateWithFormat:predicateFormat arguments:args]];
 }
 
+- (id)valueForKeyPath:(NSString *)keyPath {
+    if (_backingArray) {
```

Comment:
So you'd early return if there is no backing array by calling through to `super`? That would at least save us one level in nesting and is more similar to the rest of the methods. :+1: